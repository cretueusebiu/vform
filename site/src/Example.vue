<template>
  <div>
    this is an Examples
    <button @click="click">
      click
    </button>
  </div>
</template>

<script>
export default {
  methods: {
    click () {
      alert('clicked')
    }
  }
}
</script>
