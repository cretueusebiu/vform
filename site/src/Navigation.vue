<template>
  <nav aria-label="Navigation">
    <a v-for="item in items"
       :key="item.title"
       :href="`#${item.title.toLocaleLowerCase().replaceAll(' ', '-')}`"
       :class="[
         false ? 'bg-gray-200 text-gray-900' : 'text-gray-600 hover:bg-gray-50 hover:text-gray-900',
         item.sub ? '' : 'font-medium',
         'flex items-center px-3 py-2 rounded-md'
       ]"
    >
      <span class="truncate">
        {{ item.title }}
      </span>
    </a>
  </nav>
</template>

<script>
export default {
  data: () => ({
    items: [
      { title: 'Getting Started' },
      { title: 'Installation', sub: true },
      { title: 'Basic Usage', sub: true },
      { title: 'Examples' }
    ]
  })
}
</script>
