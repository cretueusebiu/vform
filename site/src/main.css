@tailwind base;
@tailwind components;
@tailwind utilities;

pre[class*="language-"] {
  @apply p-3;
}
