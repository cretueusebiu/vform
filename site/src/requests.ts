import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

const mock = new MockAdapter(axios, { delayResponse: 500 })

mock.onPost('/login-invalid').reply(422, {
  errors: {
    email: ['The email field is invalid.'],
    password: ['The password field is invalid.']
  },
  message: 'The given data was invalid.'
})
