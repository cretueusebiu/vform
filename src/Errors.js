export default class Errors {
  /**
   * Create a new error bag instance.
   */
  constructor () {
    this.errors = {}
  }

  /**
   * Set the errors object.
   *
   * @param {Object}
   */
  set (errors) {
    this.errors = errors
  }

  /**
   * Get all the errors.
   *
   * @return {Object}
   */
  all () {
    return this.errors
  }

  /**
   * Determine if there is an error for the given field.
   *
   * @param  {String} field
   * @return {Boolean}
   */
  has (field) {
    return this.errors.hasOwnProperty(field)
  }

  /**
   * Determine if there are any errors for the given fields.
   *
   * @param  {...String} fields
   * @return {Boolean}
   */
  hasAny (...fields) {
    return fields.some(field => this.has(field))
  }

  /**
   * Determine if there are any errors.
   *
   * @return {Boolean}
   */
  any () {
    return Object.keys(this.errors).length > 0
  }

  /**
   * Get the error message for the given field.
   *
   * @param  {String} field
   * @return {String|undefined}
   */
  get (field) {
    if (this.has(field)) {
      return this.list(field)[0];
    }
  }
  
  /**
   * Get the array of all error messages for the given field.
   *
   * @param  {String} field
   * @return {Array}
   */
  list (field) {
    if (this.has(field)) {
      const messages = this.errors[field]
      
      return Array.isArray(messages) ? messages : [messages]
    }
  }

  /**
   * Get the error message for the given fields.
   *
   * @param  {...String} fields
   * @return {Array}
   */
  only (...fields) {
    const messages = []

    fields.forEach(field => {
      const message = this.get(field)

      if (message) {
        messages.push(message)
      }
    })

    return messages
  }

  /**
   * Get all the errors in a flat array.
   *
   * @return {Array}
   */
  flatten () {
    return Object.values(this.errors).reduce((a, b) => a.concat(b), [])
  }

  /**
   * Clear one or all error fields.
   *
   * @param {String|undefined} field
   */
  clear (field) {
    const errors = {}

    if (field) {
      Object.keys(this.errors).forEach(key => {
        if (key !== field) {
          errors[key] = this.errors[key]
        }
      })
    }

    this.set(errors)
  }
}
