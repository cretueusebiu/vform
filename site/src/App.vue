<template>
  <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
    <div class="max-w-5xl mx-auto">
      <div class="mt-4 flex items-start">
        <div>
          <div class="text-2xl font-normal text-indigo-600">
            <a href="/">vform</a>
          </div>
          <div class="mt-1 dark:text-gray-300">
            Handle Laravel-Vue forms and validation with ease.
          </div>
        </div>

        <ToggleTheme class="ml-auto my-auto" />
      </div>

      <Badges class="mt-2" />

      <div class="grid grid-cols-4 gap-5 mt-10">
        <div class="col-span-1">
          <Navigation />
        </div>

        <div class="col-span-3">
          <Demo class="mb-4" />

          <div class="panel">
            <Docs />
          </div>
        </div>
      </div>

      <div class="text-center py-4 text-gray-400">
        Created by <a href="https://github.com/cretueusebiu" target="_blank">Cretu Eusebiu</a> âœŒ
      </div>
    </div>
  </div>
  </div>
</template>

<script lang="ts">
import Docs from './../../docs.md'
import Demo from './components/Demo.vue'
import Badges from './components/Badges.vue'
import Navigation from './components/Navigation.vue'
import ToggleTheme from './components/ToggleTheme.vue'

export default {
  name: 'App',

  components: {
    Docs,
    Demo,
    Badges,
    Navigation,
    ToggleTheme
  }
}
</script>
