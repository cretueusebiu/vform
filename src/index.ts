import Form from './Form'
import Errors from './Errors'

export {
  Form,
  Errors,
  Form as default
}
