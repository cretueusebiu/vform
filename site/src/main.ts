import Vue from 'vue'
import App from './App.vue'
import './main.css'

Vue.config.productionTip = false

new Vue({
  render: h => h(App)
}).$mount('#app')
