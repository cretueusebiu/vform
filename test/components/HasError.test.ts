import { mount } from '@vue/test-utils'
import { Form, HasError } from './../../src'

describe('HasError', () => {
  test('render error message', () => {
    const form = new Form({ username: '' })
    form.errors.set({ username: 'Username is required' })

    const wrapper = mount(HasError, {
      propsData: { form, field: 'username' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('doesn`t render error message', () => {
    const form = new Form({ username: '' })

    const wrapper = mount(HasError, {
      propsData: { form, field: 'username' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
