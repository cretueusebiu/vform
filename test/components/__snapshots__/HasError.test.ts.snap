// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HasError doesn\`t render error message 1`] = `""`;

exports[`HasError render error message 1`] = `<div class="help-block invalid-feedback">Username is required</div>`;
